extension/parquet/reader/variant/variant_shredded_conversion.cpp

```diff
@@ -0,0 +1,553 @@
+			throw NotImplementedException("Decimal with PhysicalType (%s) not implemented for shredded Variant",
+			                              EnumUtil::ToString(physical_type));
+		}
+	}
+	//! date
+	case LogicalTypeId::DATE: {
+		return ConvertTypedValues<date_t, ConvertShreddedValue<date_t>, LogicalTypeId::DATE>(
+		    typed_value, metadata, value, offset, length, total_size);
+	}
+	//! time
+	case LogicalTypeId::TIME: {
+		return ConvertTypedValues<dtime_t, ConvertShreddedValue<dtime_t>, LogicalTypeId::TIME>(
+		    typed_value, metadata, value, offset, length, total_size);
+	}
+	//! timestamptz(6) (timestamptz(9) not implemented in DuckDB)
+	case LogicalTypeId::TIMESTAMP_TZ: {
+		return ConvertTypedValues<timestamp_t, ConvertShreddedValue<timestamp_t>, LogicalTypeId::TIMESTAMP_TZ>(
+		    typed_value, metadata, value, offset, length, total_size);
+	}
+	//! timestampntz(6)
+	case LogicalTypeId::TIMESTAMP: {
+		return ConvertTypedValues<timestamp_t, ConvertShreddedValue<timestamp_t>, LogicalTypeId::TIMESTAMP>(
+		    typed_value, metadata, value, offset, length, total_size);
+	}
+	//! timestampntz(9)
+	case LogicalTypeId::TIMESTAMP_NS: {
+		return ConvertTypedValues<timestamp_ns_t, ConvertShreddedValue<timestamp_ns_t>, LogicalTypeId::TIMESTAMP_NS>(
+		    typed_value, metadata, value, offset, length, total_size);
+	}
+	//! binary
+	case LogicalTypeId::BLOB: {
+		return ConvertTypedValues<string_t, ConvertShreddedValue<string_t>, LogicalTypeId::BLOB>(
+		    typed_value, metadata, value, offset, length, total_size);
+	}
+	//! string
+	case LogicalTypeId::VARCHAR: {
+		return ConvertTypedValues<string_t, ConvertShreddedValue<string_t>, LogicalTypeId::VARCHAR>(
+		    typed_value, metadata, value, offset, length, total_size);
+	}
+	//! uuid
+	case LogicalTypeId::UUID: {
+		return ConvertTypedValues<hugeint_t, ConvertShreddedValue<hugeint_t>, LogicalTypeId::UUID>(
+		    typed_value, metadata, value, offset, length, total_size);
+	}
+	default:
+		throw NotImplementedException("Variant shredding on type: '%s' is not implemented", type.ToString());
+	}
+}
+
+namespace {
+
+struct ShreddedVariantField {
+public:
+	explicit ShreddedVariantField(const string &field_name) : field_name(field_name) {
+	}
+
+public:
+	string field_name;
+	//! Values for the field, for all rows
+	vector<VariantValue> values;
+};
+
+} // namespace
+
+template <bool IS_REQUIRED>
+static vector<VariantValue> ConvertBinaryEncoding(Vector &metadata, Vector &value, idx_t offset, idx_t length,
+                                                  idx_t total_size) {
+	UnifiedVectorFormat value_format;
+	value.ToUnifiedFormat(total_size, value_format);
+	auto value_data = value_format.GetData<string_t>(value_format);
+	auto &validity = value_format.validity;
+
+	UnifiedVectorFormat metadata_format;
+	metadata.ToUnifiedFormat(length, metadata_format);
+	auto metadata_data = metadata_format.GetData<string_t>(metadata_format);
+
+	vector<VariantValue> ret(length);
+	if (IS_REQUIRED) {
```

Comment:
This might be overkill, `AllValid()` should already cover this.
It was mostly added because during development this was expected to *always* be required, but object fields caused this to change:
> A field's value and typed_value are set to null (missing) to indicate that the field does not exist in the variant

`ConvertBinaryEncoding` is used only when `typed_value` is entirely missing for the field, but because the field can be missing, the `value` for the field can contain nulls - this is illegal in all other contexts.